src/java/map/ConcurrentFloatMask.java

```diff
@@ -97,18 +98,48 @@ public ConcurrentFloatMask clampMax(float value) {
         );
     }
 
+    public ConcurrentFloatMask clampMaxInArea(float value, ConcurrentBinaryMask area) {
+        return Pipeline.add(this, Arrays.asList(this, area), res ->
+                this.floatMask.clampMaxInArea(value, ((ConcurrentBinaryMask) res.get(1)).getBinaryMask())
+        );
+    }
+
     public ConcurrentFloatMask clampMin(float value) {
         return Pipeline.add(this, Collections.singletonList(this), res ->
                 this.floatMask.clampMin(value)
         );
     }
 
+    public ConcurrentFloatMask clampMinInArea(float value, ConcurrentBinaryMask area) {
+        return Pipeline.add(this, Arrays.asList(this, area), res ->
+                this.floatMask.clampMinInArea(value, ((ConcurrentBinaryMask) res.get(1)).getBinaryMask())
+        );
+    }
+
     public ConcurrentFloatMask threshold(float value) {
         return Pipeline.add(this, Collections.singletonList(this), res ->
                 this.floatMask.threshold(value)
         );
     }
 
+    public ConcurrentFloatMask useBrush(Vector2f location, String brushName, float intensity, int size) {
+        return Pipeline.add(this, Collections.singletonList(this), res ->
+                this.floatMask.useBrush(location, brushName, intensity, size)
+        );
+    }
```

Comment:
This function most likely wont work with concurrent masks if it needs a location from another mask as the location wont be defined when this is called.